python/ray/data/dataset_pipeline.py

```diff
@@ -1072,6 +1072,15 @@ def iter_torch_batches(
             local_shuffle_seed=local_shuffle_seed,
         )
 
+    def _is_tensor_dataset(self) -> bool:
+        """Return ``True`` if this dataset is a tensor dataset."""
+        from ray.air.constants import TENSOR_COLUMN_NAME
+
+        schema = self.schema()
+        if schema is None or isinstance(schema, type):
+            return False
+        return schema.names == [TENSOR_COLUMN_NAME]
```

Comment:
@clarkzinzow would it make sense to consolidate `TENSOR_COLUMN_NAME` and `VALUE_COL_NAME` into a single constant? It seems like they represent the same thing.

https://github.com/ray-project/ray/blob/f05c5f96b7c715a6220043845183d124f64a7b45/python/ray/data/_internal/table_block.py#L15-L17

https://github.com/ray-project/ray/blob/f05c5f96b7c715a6220043845183d124f64a7b45/python/ray/air/constants.py#L19-L20

@stephanie-wang I'm wondering if we could abstract this logic into a function and then re-use it across Data and AIR. I've seen it re-implemented across several functions and methods, and I'm worried about code duplication. What do you think?

Reply:
It would definitely make sense to consolidate those into a single constant (I actually wasn't aware that there are two!); we could standardize on the `ray.air.constants.TENSOR_COLUMN_NAME`.

As for abstracting this logic, we've been sharing this logic within Datasets via `TableBlockAccessor.is_tensor_wrapper()`, since the Datasets data model is centralized on these blocks. https://github.com/ray-project/ray/blob/f05c5f96b7c715a6220043845183d124f64a7b45/python/ray/data/_internal/table_block.py#L159-L160

Having a single shared utility for use across Datasets and AIR makes sense to me; it would only need to explicitly check Pandas DataFrames and Arrow Tables, returning `False` for everything else, and `TableBlockAccessor.is_tensor_wrapper()` could just use that utility under-the-hood.